src/Admin/ReaderThemes.php

```diff
@@ -210,17 +227,38 @@ public function get_default_reader_themes() {
 				]
 			);
 
-			if ( ! is_wp_error( $response ) ) {
-				set_transient( $cache_key, $response, DAY_IN_SECONDS );
+			if ( is_array( $response ) ) {
+				$response = (object) $response;
 			}
-		}
 
-		if ( is_wp_error( $response ) ) {
-			return [];
-		}
+			/**
+			 * The response must minimally be an object with a themes array.
+			 *
+			 * @see https://wordpress.org/support/topic/issue-during-activating-the-updated-plugins/#post-13383737
+			 */
+			if (
+				! is_object( $response )
+				|| ! property_exists( $response, 'themes' )
+				|| ! is_array( $response->themes )
+				|| is_wp_error( $response )
```

Comment:
If `$response` is a `WP_Error` (and `WP_DEBUG_DISPLAY` is enabled) why not set it to be `$this->themes_api_error` here? Or at least use the message in the `amp_themes_api_invalid_response` error?

Reply:
Done in 0e13feccc